### ISSUE TITLE
> I don't love this solution, but it does fix the Windows VMSS issue. I'd prefer to see a password only get set on create, rather than on update too.

### ISSUE THREAD

**Issue by CecileRobertMichon** [kubernetes-sigs/cluster-api-provider-azure#1190]
> I don't love this solution, but it does fix the Windows VMSS issue. I'd prefer to see a password only get set on create, rather than on update too.
> 
> In the upcoming MachinePoolMachine changes, the hash is deprecated in favor of a model diff. I do fear this issue will still be a problem if the OsProfile continues to mutate.

Agreed. Let's follow up on this.

_Originally posted by @CecileRobertMichon in https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1186#issuecomment-783635800_

**Comment by jsturtevant** [comment]
agreed it isn't ideal.  Right now Cloudbase-init is configured to reset the password on the machine to a random password on first boot but the VM does need to be provided a temp password for creation.  This was to make it so the customers don't have to have a "magic" windows password which they have to manage out of band.

Considered a few couple other options in the PR to fix the hash generation:

- Introduce a Windows Password to the spec to store the password.  
  - Decided against this because we wanted to avoid encouraging "magic" windows password, plus it is reset via cloudbase init.
- Generate the temp password deterministically based of some properties of the template. 
  - Though only temporary and unlikely it could be used, thought it was better to have a random password for that temporary time period.
